src/core/devtools/widget/module_list.h

```diff
@@ -0,0 +1,89 @@
+//  SPDX-FileCopyrightText: Copyright 2024 shadPS4 Emulator Project
+//  SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include <algorithm>
+#include <filesystem>
+#include <mutex>
+#include <string>
+#include <vector>
+
+#include "common/path_util.h"
+
+namespace Core::Devtools::Widget {
+
+class ModuleList {
+public:
+    ModuleList() = default;
+    ~ModuleList() = default;
+
+    void Draw();
+    bool open = false;
+
+    static std::filesystem::path game_folder;
+
+    static void SetGameFolder(const std::filesystem::path& path) {
+        game_folder = path;
+    }
+
+    static bool IsSystemModule(const std::filesystem::path& path) {
+
+        const auto sys_modules_path = Common::FS::GetUserPath(Common::FS::PathType::SysModuleDir);
+
+        const auto canonical_path = std::filesystem::canonical(path);
+        const auto canonical_sys_path = std::filesystem::canonical(sys_modules_path);
+
+        const auto path_str = canonical_path.string();
+        const auto sys_path_str = canonical_sys_path.string();
+
+        return path_str.starts_with(sys_path_str);
+    }
+
+    static bool IsSystemModule(const std::string& name) {
+        const auto game_modules_path = game_folder / "sce_module";
+        const auto prx_path = game_modules_path / name;
+
+        if (!std::filesystem::exists(prx_path)) {
+            return true;
+        }
+        return false;
+    }
+
+    static void AddModule(const std::string& name, std::filesystem::path path) {
+        if (name == "eboot.bin") {
+            return;
+        }
+        std::scoped_lock lock(modules_mutex);
+        modules.push_back({name, IsSystemModule(path), true});
+    }
+
+    static void AddModule(std::string& name) {
+        name = name + ".prx";
```

Comment:
Don't edit reference parameters 

```suggestion
    static void AddModule(std::string name) {
        name = name + ".prx";
```